docs/content/en/docs/concepts/_index.md

```diff
@@ -179,3 +179,26 @@ Skaffold provides two separate operating modes:
 
 Skaffold command-line interface also provides other functionalities that may
 be helpful to your project. For more information, see [CLI References](/docs/references/cli).
+
+## Local development
+
+Local development means that Skaffold can skip pushing built container images, because the images are already present where they are run.
+For standard development setups such as `minikube` and `docker-for-desktop`, this works out of the box.
+
+However, for non-standard local setups, such as `minikube` profiles, some extra configuration is necessary.
+The essential steps are:
+
+1. Ensure that Skaffold builds the images with the docker daemon, which also runs the containers.
+2. Tell Skaffold to skip pushing images.
```

Comment:
Maybe add this as a step 3 below?

3. Tell skaffold to skip pushing images by adding `push:false` to your skaffold config:
```yaml
build:
  local:
   push: false
```

Reply:
Good point. This is should be mentioned as an alternative to the `skaffold config set -k context local-cluster true` setting.